scikits/bootstrap/bootstrap.py

```diff
@@ -202,9 +214,11 @@ def ci(data: Union[Tuple[np.ndarray, ...], np.ndarray],
         return _ci_abc(tdata, statfunction, epsilon, alphas, output, multi)
 
     if multi != 'independent':
-        bootindices = bootstrap_indices(tdata[0], n_samples)
-        stat = np.array([statfunction(*(x[indices] for x in tdata))
-                            for indices in bootindices])
+        if statfunction in (np.average, np.mean) and len(tdata) == 1:
+            stat = _calculate_boostrap_mean_stat(tdata[0], n_samples)
+        else:
+            rng = np.random.default_rng()
+            stat = np.array([statfunction(*(rng.choice(x, tdata[0].shape[0]) for x in tdata)) for _ in range(n_samples)])
```

Comment:
This is not entirely equivalent, I think.  When multi != 'independent', the arrays in tdata should be treated as being paired: the *same* samples should be taken for each of them.  Here, it looks like rng.choice is called separately for each.

Reply:
I need to cover this with tests to be sure that it works in the same way.